pivy-importer/src/main/groovy/com/linkedin/python/importer/pypi/PypiApiCache.groovy

```diff
@@ -26,8 +27,17 @@ class PypiApiCache {
 
     Map<String, ProjectDetails> cache = [:].withDefault { String name -> new ProjectDetails(downloadMetadata(name)) }
 
-    ProjectDetails getDetails(String project) {
-        return cache.get(project)
+    ProjectDetails getDetails(String project, boolean lenient) {
+        try {
+            return cache.get(project)
+        } catch(HttpResponseException httpResponseException) {
```

Comment:
It might be all kinds of error responses, https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/http/HttpResponseException

Currently this is the only one we have seen, maybe there is a change server throws "internal server error" or other error code, we don't know, as long as we haven't seen others, we suggest we default it to illegal project name, which is more than 90% cases (might be more)

Reply:
Ok. Could we append the error message from the exception in case it isn't covered by the 90%+ case so we can at least get some meaningful info from it?

Reply:
Good idea, we are already doing it.